fv3gfs-util/fv3gfs/util/partitioner.py

```diff
@@ -723,24 +727,49 @@ def tile_extent_from_rank_metadata(
 
 
 def rank_extent_from_tile_metadata(
-    dims: Sequence[str], tile_extent: Sequence[int], layout: Tuple[int, int]
-) -> Tuple[int, ...]:
+        dims: Sequence[str], tile_extent: Sequence[int], subtile_index: Tuple[int, int],
```

Comment:
If all the current tests pass without this change, then it means scatter/gather only breaks when `edge_tile_ratio` on the TilePartitioner is not 1. That would mean we can consider merging this code first, then adding a test combining these two features (which would fail) and fixing the code in another PR.

Reply:
Since you changed the call signature to this routine, could you add new test cases that make use of the new arguments and refactor the existing tests to pass (i.e. provide these new required arguments)?

Reply:
Done. Scatter/Gather seem to work with an unchanged `edge_interior_ratio` as all the current tests are passing.